### ISSUE TITLE
question:UnicodeDecodeError at /admin/blog/author/import/

### ISSUE THREAD

**Issue by hebijiandai** [django-import-export/django-import-export#72]
Sir,thanks a lot to develop this great repository.
I have  little problem to solve , as below
the code:
```python
#in models.py
class Author(models.Model):
    author = models.CharField('作者', max_length=50)
    title = models.CharField('标题', max_length=150)
    qualification = models.ForeignKey(Qualification)
    mark = models.ManyToManyField(Mark)
    blog = models.TextField('博客内容')
    time = models.DateField('写作日期')

    def __unicode__(self):
        return unicode(self.author)

    class Meta:
        ordering = ['time']
```
```python
#in admin.py
class AuthorAdmin(ImportExportModelAdmin, admin.ModelAdmin):
    search_fields = ('author', 'title', 'mark', 'blog')
    list_display = ('author', 'title', 'time')
```
When I export the data,
![1](https://f.cloud.github.com/assets/5127081/2341019/59bfb640-a4cf-11e3-8ad6-ce0e2a5172f2.png)
but the csv file appear to be the retortion,
![2](https://f.cloud.github.com/assets/5127081/2341031/9691934a-a4cf-11e3-84d1-511eee6f4839.png)
when I modified the second row author data and import,
 
![3](https://f.cloud.github.com/assets/5127081/2341044/d08317ea-a4cf-11e3-84d2-dae694abc8d1.png)
it cause the error,
![4](https://f.cloud.github.com/assets/5127081/2341046/e01c5892-a4cf-11e3-8136-623c4037c9b2.png)

how I it be smoothly modified and import？thanks.
                                                                                       Allenlee

**Comment by pawelnowak** [comment]
Hi guys,

I've stumbled upon the exact same issue in one of my company's projects. We're using python3 and I'm 100% sure there's no **u'** missing on any model's __ str__ method. Yet, the UnicodeDecodeError exception still occurs when trying to import a csv file containing utf-8 encoded characters in the django admin.

It turns out that adding the **encoding** param with value of **utf-8** to the TempFolderStorage.open's method seems to solve the problem, at least in our environment.

Here's a quick'n'dirty fix in case anyone googles this issue trying to find a solution, like I did a few hours ago.

First we subclass the TempFolderStorage class, adding said param:
```python
import tempfile
from import_export.tmp_storages import TempFolderStorage

class Utf8TempFolderStorage(TempFolderStorage):

    def open(self, mode='r'):
        if self.name:
            return open(self.get_full_path(), mode, encoding='utf-8')
        else:
            tmp_file = tempfile.NamedTemporaryFile(delete=False)
            self.name = tmp_file.name
            return tmp_file
```
Then we point the new class as IMPORT_EXPORT_TMP_STORAGE_CLASS in settings.py:
```python
 IMPORT_EXPORT_TMP_STORAGE_CLASS = 'path.to.storages.Utf8TempFolderStorage'
```
